torchdynamo/symbolic_convert.py

```diff
@@ -397,6 +397,24 @@ def call_function(self, fn, args, kwargs):
                     self.push(ConstantVariable(len(item), **options))
                 else:
                     unimplemented(f"`len` with arg type {arg}")
+            elif fn.fn is isinstance:
+                assert not kwargs and len(args) == 2
+                arg, *isinstance_types = args
+                get_type = (
+                    lambda x: x.fn if isinstance(x, BuiltinVariable) else x.as_proxy()
+                )
+                isinstance_types = tuple(map(get_type, isinstance_types))
+                if isinstance(arg, TensorVariable):
+                    self.push(
+                        BasicTypeVariable(
+                            self.create_proxy(
+                                "call_function", len, (arg.as_proxy(),), {}
```

Comment:
`len` seems wrong here?

We should have a special case for `isinstance(, Tensor)` that results in a constant.  Since we guard on types, in the common case we don't actually need to run isinstance() as part of the graph.  The guard is checking the type of `x` so checking it again is redundant.